src/api/middleware/auth.js

```diff
@@ -48,9 +57,40 @@ const isAdmin = (req, res, next) => {
     return next();
 };
 
+const isAdminOrGod = (req, res, next) => {
+    if (!req?.user?.isAdmin) {
+        return isGod(req, res, next);
+    }
+
+    return next();
+};
+
+const isOfferOwner = (offerId) => async (req, res, next) => {
+
+    let offer;
+    try {
+        offer = await (new OfferService()).getOfferById(offerId, req.user);
+    } catch (err) {
+        console.error(err);
+        throw err;
+    }
+    if (offer.owner.toString() !== req.user?.company?._id.toString() &&
+                req.body.god_token !== config.god_token &&
+                !req.user?.isAdmin) {
+        return res.status(HTTPStatus.FORBIDDEN).json({
+            reason: ValidationReasons.NOT_OFFER_OWNER(offerId),
+            error_code: ErrorTypes.FORBIDDEN,
+        });
+    }
+    return next();
```

Comment:
```suggestion
    try {
        const offer = await (new OfferService()).getOfferById(offerId, req.user);
        
        if (offer.owner.toString() !== req.user?.company?._id.toString() &&
                req.body.god_token !== config.god_token &&
                !req.user?.isAdmin) {
        return res.status(HTTPStatus.FORBIDDEN).json({
            reason: ValidationReasons.NOT_OFFER_OWNER(offerId),
            error_code: ErrorTypes.FORBIDDEN,
        });
    }
    return next();
    } catch (err) {
        console.error(err);
        throw err;
    }
    
```

I'm not sure how the code was before when I told you to separate it to another try/catch, but it's really not necessary, since inside the verifications are returns, not throws, so the catch would only trigger when the offer does not exist. 

Next time you can refute if you notice stuff like this :P

Reply:
After seeing another comment on this, I understood. So yep, you can use only one try catch as mentioned above, and keep the return, instead of the throw you had before, exactly like the suggestion

Reply:
So should I just keep the changes I made or revert them? I'm a little bit confused

Reply:
just apply the suggested changes, there's a mix of keeping stuff and applying new things

Before, you had a try with different throws inside, that's why I told you to separate them. Now, since there's only one throw, there's no need for separation, hence the 1 try/catch with the throw and return inside